server/handlers/component_handler.go

```diff
@@ -55,12 +55,12 @@ func (h *Handler) GetMeshmodelModelsByCategories(rw http.ResponseWriter, r *http
 	}
 	pagestr := queryParams.Get("page")
 	page, _ := strconv.Atoi(pagestr)
-	if page <= 0 {
```

Comment:
Replace the repetitive logic for extracting `page, pagesize, search, offset` from API handlers to use  `getPaginationParams ` in `utils.go`.